### PR TITLE
LLEXT: convert most remaining Module Adapter drivers

### PR DESCRIPTION
**Pull request by lyakh**
This converts most remaining Module Adapter drivers to LLEXT modules without activating modular build for them. This also means that these modules haven't been tested for loading and therefore can use unexported symbols, in which case suitable `EXPORT_SYMBOL()` calls will have to be added

### CODE REVIEW

File: src/audio/codec/dts/llext/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2024 Intel Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_DTS_CODEC_STUB)
+sof_llext_build("dts"
+	SOURCES ../dts.c
+		../dts_stub.c
+)
+target_include_directories(dts_llext_lib PRIVATE
+	"../../../../../third_party/include"
```
Comment (lgirdwood) [inline src/audio/codec/dts/llext/CMakeLists.txt:10]: I assume we cant use ```${SOME_PATH}``` here today ?

**Review by lgirdwood — APPROVED** [review]
A future update could fix the include paths to remove the ../../../

File: src/audio/crossover/crossover.c [hunk 2]
```diff
@@ -636,3 +636,22 @@ static const struct module_interface crossover_interface = {
 
 DECLARE_MODULE_ADAPTER(crossover_interface, crossover_uuid, crossover_tr);
 SOF_MODULE_INIT(crossover, sys_comp_module_crossover_interface_init);
+
+#if CONFIG_COMP_CROSSOVER_MODULE
+/* modular: llext dynamic link */
+
+#include <module/module/api_ver.h>
+#include <module/module/llext.h>
+#include <rimage/sof/user/manifest.h>
```
Comment (cujomalainey) [inline src/audio/crossover/crossover.c:645]: the crossover depends on math iir. Is there any risk of breaking that here?

File: src/audio/codec/dts/llext/CMakeLists.txt (hunk 1 above)
Comment (lyakh) [reply src/audio/codec/dts/llext/CMakeLists.txt:10]: @lgirdwood not sure, need to try

File: src/audio/crossover/crossover.c (hunk 2 above)
Comment (lyakh) [reply src/audio/crossover/crossover.c:645]: @cujomalainey as long as iir math is built into the image - no, but if / when we make it modular too, then we'll need to make sure to build and load it to satisfy the dependency

**Comment by lyakh** [comment]
@wszypelt as far as I can see two TGL QB failures cannot be related?

File: app/boards/intel_adsp_ace15_mtpm.conf [hunk 3]
```diff
@@ -90,6 +90,10 @@ CONFIG_LLEXT_STORAGE_WRITABLE=y
 CONFIG_MODULES=y
 CONFIG_LIBRARY_BASE_ADDRESS=0xa0688000
 
+# Modular by default components. Compile-tested but usually not needed at run-time
```
Comment (kv2019i) [inline app/boards/intel_adsp_ace15_mtpm.conf:95]: Hmm, why are Intel board file changes and generic module code changes in the same commit? Especially as the config change is done to only some targets (e.g. not even all ace targets), so a commit explaining the reasoning would be nice.

**Review by kv2019i — COMMENTED** [review]
See question inline about putting board file changes togehter with module changes.

File: src/audio/crossover/crossover.c (hunk 2 above)
Comment (lgirdwood) [reply src/audio/crossover/crossover.c:645]: @cujomalainey next steps are to make maths library (and other utility code) into per feature "modules" and load into SRAM on demand i.e. minimize SRAM use for power and for active modules only, further along the road is using llext to put hot (time sensitive) code in SRAM and cold code in DRAM to further optimize/detereminize processing performance.

**Comment by lgirdwood** [comment]
@lyakh jenkins looks fine, the not runs are due to v2.11 stressing runs.

File: src/audio/codec/dts/llext/CMakeLists.txt (hunk 1 above)
Comment (lyakh) [reply src/audio/codec/dts/llext/CMakeLists.txt:10]: done

**Review by lgirdwood — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Pull request merged**